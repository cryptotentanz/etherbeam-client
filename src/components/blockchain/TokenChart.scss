@use '../../styles/resources/colors';
@use '../../styles/resources/sizes';

.my-tokenchart-unavailable {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: colors.$disabled-fg;
  font-size: sizes.$size-xl;
  font-weight: bold;
  font-style: italic;
}
