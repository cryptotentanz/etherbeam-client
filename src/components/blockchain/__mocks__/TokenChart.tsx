import React, { FC } from 'react'

const TokenChart: FC = () => {
  return <div></div>
}

export default TokenChart
