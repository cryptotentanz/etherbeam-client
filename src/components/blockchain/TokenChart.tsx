import React, { CSSProperties, FC } from 'react'
import { faChartBar } from '@fortawesome/free-solid-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import TradingViewWidget, { Themes } from 'react-tradingview-widget'
import './TokenChart.scss'

interface TokenChartProps {
  pair: string | null | undefined
  width?: number
  height?: number
}

const TokenChart: FC<TokenChartProps> = ({ pair, width, height }: TokenChartProps) => {
  const classNames = [
    (!width && 'w-100') || '',
    (!height && 'h-100') || '',
    !pair ? 'my-tokenchart-unavailable' : '',
  ].join(' ')
  const style: CSSProperties = {
    width: width ? `${width}px` : 'auto',
    height: height ? `${height}px` : 'auto',
  }

  return (
    <div className={classNames} style={style}>
      {pair ? (
        <TradingViewWidget
          symbol={`UNISWAP:${pair}`}
          interval={60}
          timezone="Europe/Paris"
          theme={Themes.DARK}
          allow_symbol_change={false}
          show_popup_button
          withdateranges
          autosize
        />
      ) : (
        <>
          <FontAwesomeIcon icon={faChartBar} size="5x" />
          <span>No chart available</span>
        </>
      )}
    </div>
  )
}

export default TokenChart
