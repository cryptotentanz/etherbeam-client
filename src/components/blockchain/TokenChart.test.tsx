import * as React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'

jest.mock('react-tradingview-widget', () => {
  return {
    __esModule: true,
    // eslint-disable-next-line react/display-name
    default: () => {
      return <article></article>
    },
    Themes: {
      DARK: 'DARK',
    },
  }
})

import TokenChart from './TokenChart'

describe('<TokenChart />', () => {
  let component: ShallowWrapper

  describe('Default', () => {
    beforeAll(() => {
      component = shallow(<TokenChart pair="UNIWETH" />)
    })

    it('should render div', () => {
      const div = component.find('div').at(0)
      expect(div.hasClass('w-100')).toBeTruthy()
      expect(div.hasClass('h-100')).toBeTruthy()
    })

    it('should render widget', () => {
      const widget = component.find('div').at(0).children().at(0)
      expect(widget.prop('symbol')).toBe('UNISWAP:UNIWETH')
      expect(widget.prop('interval')).toBe(60)
      expect(widget.prop('timezone')).toBe('Europe/Paris')
      expect(widget.prop('allow_symbol_change')).toBeFalsy()
      expect(widget.prop('show_popup_button')).toBeTruthy()
      expect(widget.prop('withdateranges')).toBeTruthy()
      expect(widget.prop('autosize')).toBeTruthy()
    })

    it('should match snapshot', () => expect(component.render()).toMatchSnapshot())
  })

  describe('No pair', () => {
    beforeAll(() => {
      component = shallow(<TokenChart pair={null} />)
    })

    it('should render div', () =>
      expect(component.find('div').at(0).hasClass('my-tokenchart-unavailable')).toBeTruthy())
  })

  describe('With width', () => {
    beforeAll(() => {
      component = shallow(<TokenChart pair="UNIWETH" width={200} />)
    })

    it('should render div', () => {
      const div = component.find('div').at(0)
      expect(div.hasClass('w-100')).toBeFalsy()
      expect(div.hasClass('h-100')).toBeTruthy()
    })
  })

  describe('With height', () => {
    beforeAll(() => {
      component = shallow(<TokenChart pair="UNIWETH" height={200} />)
    })

    it('should render div', () => {
      const div = component.find('div').at(0)
      expect(div.hasClass('w-100')).toBeTruthy()
      expect(div.hasClass('h-100')).toBeFalsy()
    })
  })

  describe('With width height', () => {
    beforeAll(() => {
      component = shallow(<TokenChart pair="UNIWETH" width={200} height={200} />)
    })

    it('should render div', () => {
      const div = component.find('div').at(0)
      expect(div.hasClass('w-100')).toBeFalsy()
      expect(div.hasClass('h-100')).toBeFalsy()
    })
  })
})
