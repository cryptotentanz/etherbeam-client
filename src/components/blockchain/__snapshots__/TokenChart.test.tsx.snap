// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TokenChart /> Default should match snapshot 1`] = `
<div
  class="w-100 h-100 "
  style="width:auto;height:auto"
>
  <article />
</div>
`;
